Disable Style/NegatedIf cop + upgrade to Rails 6.1.7 (bundle audit)

sebastienhoorens opened the pull request:
I also made an example change that I think makes more sense now, to test that CI still passes. Double negation is still prevented by other cops:

<img width="785" alt="Screenshot 2023-01-20 at 10 03 07" src="https://user-images.githubusercontent.com/31925816/213658970-b65f5cc2-d8d4-4763-a400-215d86cff475.png">
<img width="736" alt="Screenshot 2023-01-20 at 10 02 14" src="https://user-images.githubusercontent.com/31925816/213658981-2696b0c0-11c2-4d33-8b31-3d06baf2f990.png">

sebastienhoorens committed:
disable negated if cop + example change

sebastienhoorens committed:
fixing bundle audits

sebastienhoorens committed:
upgrade to rails 6.1.7

adessy commented on back/engines/commercial/flag_inappropriate_content/app/models/flag_inappropriate_content/inappropriate_content_flag.rb:
```diff
@@ -46,7 +46,7 @@ def reason_code
 
     def remove_notifications
       notifications.each do |notification|
-        unless notification.update inappropriate_content_flag: nil
```
Personally, I prefer the `unless` version but I can live with it :)

sebastienhoorens committed:
downgrading nokogiri back to 1.13.10

adessy committed:
[NO-TICKET] Fix broken tests after rails upgrade

It looks like there was a problem with how the email's body was being parsed by Nokogiri in specs, possibly due to an encoding issue. I had to change the instructions for parsing the body to make sure Nokogiri could properly handle the 7-bit encoded email.
